### ISSUE TITLE
Ios - choosing an account gets to Google home page ?!

### ISSUE THREAD

**Issue by JerryBels** [EddyVerbruggen/cordova-plugin-googleplus#592]
Hi,

So when opening the plugin on Ios, I see the `Sign in with Google` page, with the instruction "Choose an account to continue tu myapp"

Below are my account list (just one). 

I click on it, and it loads for some time, then when you expect it to get back to the app with the account data, it seems that the webview simply goes to the google homepage ! I can see my account is clearly logged in the top right corner, what a relief... Well...

When clicking on the top left "Done" button, it goes back to the app with a "The user canceled the sign-in flow." error. 

... Any help please ?

**Comment by castelar007** [comment]
same issue here!! after updating to ios12.2

**Comment by c00** [comment]
I'm experiencing the same issue.

**Comment by c00** [comment]
This is a duplicate of [this issue](https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/565).

**Comment by Bramzor** [comment]
Can anyone confirm that the fix in #565 solved it?

**Comment by c00** [comment]
I don't think there is a fix in that thread, only workarounds. The workarounds work for me.

**Comment by Zvika88** [comment]
same problem here too. And I am not using the ionic deeplinks plugin mentioned in #565 above. After successfully authenticating with google on an iOS device (iPhone 7 running up to date iOS 12.2), the user is shown to a google page (kind of an in app browser) but the plugin info is not returned to the app. If you press Done on that page - the plugin returns with a "user canceled" message.

**Comment by Zvika88** [comment]
p.s. i am on Cordova 9.0 and here is the list of plugins used:

code-push 2.0.6 "CodePushAcquisition"
cordova-android-support-gradle-release 3.0.0 "cordova-android-support-gradle-release"
cordova-plugin-app-version 0.1.9 "AppVersion"
cordova-plugin-camera 4.0.3 "Camera"
cordova-plugin-code-push 1.11.18 "CodePush"
cordova-plugin-customurlscheme 4.4.0 "Custom URL scheme"
cordova-plugin-device 1.1.4 "Device"
cordova-plugin-dialogs 2.0.1 "Notification"
cordova-plugin-facebook4 4.2.1 "Facebook Connect"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-file-transfer 1.7.1 "File Transfer"
cordova-plugin-firebase 2.0.5 "Google Firebase Plugin"
cordova-plugin-geolocation 4.0.1 "Geolocation"
cordova-plugin-googleplus 5.3.0 "Google SignIn"
cordova-plugin-inappbrowser 3.0.0 "InAppBrowser"
cordova-plugin-ionic-keyboard 2.1.3 "cordova-plugin-ionic-keyboard"
cordova-plugin-splashscreen 4.0.1 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.1 "Whitelist"
cordova-plugin-x-toast 2.7.2 "Toast"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
cordova-twitter-9 0.8.5 "Cordova Twitter 9"
cordova.plugins.diagnostic 4.0.12 "Diagnostic"

**Comment by JerryBels** [comment]
Try to downgrade to 5.2.1 as said in that topic, it worked for me.

**Comment by Zvika88** [comment]
yeah going back bypasses the problem. The plugins list I posted actually shows the list after the downgrade, note that googleplus plugin 5.3.0 is used... But when using the latest 7.0.1 I get the error as everyone else.

I cannot imagine that the 7.0.x release does not work. More likely is that one of the other plugins (that other people are using too) is causing a conflict. Maybe the new Facebook4 plugin? 

Also going back means that you are not using the latest SDK , which has negative implications.  Still it is probably the easiest short term workaround, until I or someone else have the time to dig into what is going wrong here. 

For my coming product release I will split the plugins base, use 7.0.1 for android (it solves issues that 5.3.x had), but use 5.3. for ios. That sucks, of course.

**Comment by Bramzor** [comment]
I can confirm that removing the deeplink plugin 'solved' the issue for me on iOS v12.1
So in my case, a plugin that catches the deeplink (oauth feedback) causes this specific issue.

My plugin list without deeplinks plugin:
`cordova plugins ls
cordova-clipboard 1.2.1 "Clipboard"
cordova-custom-config 5.0.2 "cordova-custom-config"
cordova-plugin-android-permissions 1.0.0 "Permissions"
cordova-plugin-camera 4.0.3 "Camera"
cordova-plugin-contacts 3.0.1 "Contacts"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-facebook4-no-zxing 4.2.2 "Facebook Connect"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-file-transfer 1.7.1 "File Transfer"
cordova-plugin-globalization 1.11.0 "Globalization"
cordova-plugin-googleplus 7.0.1 "Google SignIn"
cordova-plugin-inappbrowser 3.0.0 "InAppBrowser"
cordova-plugin-ionic 5.2.7 "cordova-plugin-ionic"
cordova-plugin-ionic-keyboard 2.1.3 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 2.3.1 "cordova-plugin-ionic-webview"
cordova-plugin-nativestorage 2.3.2 "NativeStorage"
cordova-plugin-network-information 2.0.1 "Network Information"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-sqlite-storage 2.5.1 "Cordova sqlite storage plugin"
cordova-support-google-services 1.1.0 "cordova-support-google-services"
cordova.plugins.diagnostic 4.0.10 "Diagnostic"
phonegap-plugin-barcodescanner 7.1.2 "BarcodeScanner"
phonegap-plugin-multidex 1.0.0 "Multidex"
phonegap-plugin-push 2.2.3 "PushPlugin"`

Deeplinks plugin that was removed: ionic-plugin-deeplinks 1.0.19 "Ionic Deeplink Plugin"

So next step for me is to figure out how I can still use the deeplinks plugin without it causing this plugin to spot working.
My advise is to try to figure out which plugin uses or inserts deeplinks as the facebook plugin also gets triggered by this googleplus plugin although it seems that it handles it correctly by ignoring it.
Console output shows: FB handle url: com.googleusercontent.app.....

**Comment by JerryBels** [comment]
For the record, I had no deeplink plugin in use when I got this bug. So ther eis more to it than just that.

**Comment by Bramzor** [comment]
@JerryBels, yes, that is why I mention that other plugins might also implement deeplink like functionality and get triggered by googleplus oauth like facebook for example. And probably one of them is also listening to all deeplinks which causes googleplus oauth to timeout and show the google page.

**Comment by Bramzor** [comment]
@Zvika88 : Try removing cordova-plugin-customurlscheme plugin and try without. That plugin will also  access deeplinks so might be the cause. Facebook does not trigger the issue for me.

**Comment by JerryBels** [comment]
@Bramzor Here is the list of my plugins:

```
com-sarriaroman-photoviewer 1.1.18 "PhotoViewer"
cordova-android-support-gradle-release 3.0.0 "cordova-android-support-gradle-release"
cordova-plugin-crop 0.3.1 "CropPlugin"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-facebook4 4.2.1 "Facebook Connect"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-firebase 2.0.5 "Google Firebase Plugin"
cordova-plugin-googleplus 7.0.1 "Google SignIn"
cordova-plugin-inappbrowser 3.0.0 "InAppBrowser"
cordova-plugin-ionic-keyboard 2.1.3 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.0.0 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-telerik-imagepicker 2.3.0 "ImagePicker"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-plugin-x-socialsharing 5.4.4 "SocialSharing"
cordova-wheel-selector-plugin 1.0.0 "Cordova Wheel Selector Plugin"
es6-promise-plugin 4.2.2 "Promise"
```

Gotta add a deeplink plugin again soon anyway...

**Comment by Bramzor** [comment]
For everyone using the deeplinks plugin, here is a fix that solves it for me: https://github.com/ionic-team/ionic-plugin-deeplinks/pull/194 .

**Comment by Zvika88** [comment]
> @Zvika88 : Try removing cordova-plugin-customurlscheme plugin and try without. That plugin will also access deeplinks so might be the cause. Facebook does not trigger the issue for me.

Hi @Bramzor, might be but my app needs that plugin, as I am sure most users whose apps need to run on different platforms. We all need some kind of a deeplink launch mechanism. 

The key question is whether the conflict with google login is because of some new google api or sdk changes (in which case we wont be able to overcome it and will need to stay with older googleplus), or it is something to do with the new version 7.0 of the googleplus plugin - in which case we can expect to overcome it. I hope someone on the thread can respond to that.

BTW I noticed that you are using: cordova-plugin-facebook4-no-zxing . I was able to overcome the android build "dex" problem by adding "cordova-android-support-gradle-release", you may want to take a look at that.

Cheers,
Z.

**Comment by Bramzor** [comment]
@Zvika88 Problem is related to different plugins accessing these deeplinks for different purposes. In case of facebook and google for login, but probably for other reasons as well. My advise to remove the plugin was just to make sure that it is that plugin that causes the issue. In the meantime I pushed a pull request that solves it for ionic-plugin-deeplinks. People that are not using this plugin, they have to figure out which plugin causes the issue. Or by checking CFBundleURLTypes in the plist file like I mentioned above.

**Comment by Zvika88** [comment]
@Bramzor, what was the change in the ionic-deeplinks plugin? Maybe we can instead generate a branch for the googleplus plugin that avoids the need for it. That could solve the problem in a more general way.

**Comment by Bramzor** [comment]
@Zvika88 These are the changes: https://github.com/ionic-team/ionic-plugin-deeplinks/compare/master...Bramzor:master . The biggest part is just to ignore googleplus deeplink/callback inside deeplinks to prevent that it gets triggered but the important part is commenting out line 60 which would otherwise prevent the googleplus plugin to be triggered? I'm not an object C expert but I know it's working with the code change.

**Comment by vencis** [comment]
Use this fork for Facebook instead: https://github.com/guylando/cordova-plugin-facebook4

**Comment by claudiozam** [comment]
Downgrade to 5.3.0. it worked for me.
